WalletWasabi.Fluent/ViewModels/Wallets/CoinJoinStateViewModel.cs

```diff
@@ -108,16 +108,30 @@ public CoinJoinStateViewModel(WalletViewModel walletVm, IObservable<Unit> balanc
 
 			if (_wallet.KeyManager.IsCoinjoinProfileSelected)
 			{
-				_stateMachine.Fire(Trigger.Play);
+				var overridePlebStop = _stateMachine.IsInState(State.PlebStopActive);
+				await coinJoinManager.StartAsync(_wallet, stopWhenAllMixed: !IsAutoCoinJoinEnabled, overridePlebStop, CancellationToken.None);
 			}
 		});
 
-		PauseCommand = ReactiveCommand.Create(() => _stateMachine.Fire(Trigger.Pause));
-
-		StopCommand = ReactiveCommand.Create(() => _stateMachine.Fire(Trigger.Stop));
+		StopPauseCommand = ReactiveCommand.CreateFromTask(async () =>
+		{
+			await coinJoinManager.StopAsync(_wallet, CancellationToken.None);
+		});
 
 		walletVm.Settings.WhenAnyValue(x => x.AutoCoinJoin)
-			.Subscribe(SetAutoCoinJoin);
+			.Skip(1) // The first one is triggered at the creation.
+			.SubscribeAsync(async (autoCoinJoin) =>
+			{
+				if (autoCoinJoin)
+				{
+					await coinJoinManager.StartAsync(_wallet, stopWhenAllMixed: false, false, CancellationToken.None);
+				}
+				else
+				{
+					await coinJoinManager.StopAsync(_wallet, CancellationToken.None);
```

Comment:
@danwalmsley what's the lifetime of view models? If I am coinjoining with two wallets, are there two `CoinJoinStateViewModels` viewmodels in memory subscribed to this event?

Reply:
The lifetime of `CoinJoinStateViewModel` is equal to the lifetime of the `WalletViewModel` which lifetime is equal to the application's lifetime.

Each wallet has its own `CoinJoinStateViewModel` so yes, two wallets mean two subscriptions.